src/stream.ts

```diff
@@ -27,13 +31,17 @@ export function pipeline(method: HttpMethod, uri: string | URL, options: StreamO
   const body = Utils.createBody(options.body, headers);
 
   return undici.pipeline(computedURI.url, {
-    method, headers, body, dispatcher, maxRedirections
+    method: method as HttpMethod, headers, body, dispatcher, maxRedirections
   }, ({ body }) => body);
 }
 
 export type WritableStreamCallback = (writable: Writable) => Promise<undici.Dispatcher.StreamData>;
 
-export function stream(method: HttpMethod, uri: string | URL, options: StreamOptions = {}): WritableStreamCallback {
+export function stream(
+  method: HttpMethod,
```

Comment:
```suggestion
  method: HttpMethod | WebDavMethod,
```
I'm not sure if intended to skip WevDav here ?